File: TransitionPoint.java
// this is one of the most easy approach for this question
import java.util.*;
import java.io.*;

public class Main {
 public static int transitionPoint(int arr[], int n) {
  for(int i=0;i<n;i++)
    if(arr[i] == 1)
      return i;
  return -1;
}
  public static void main(String args[]) throws IOException {
    Scanner scn = new Scanner(System.in);
    int t = scn.nextInt();
    while(t-->0)
    {
      int a = scn.nextInt();
      int[] arr = new int[a];
      for(int i=0;i<a;i++)
      {
        arr[i] = scn.nextInt();
      }
      int ans = transitionPoint(arr, a);
      System.out.println(ans);
    }
    //write your code here
    
  }
}

File: TransitionPoint.cpp
#include <bits/stdc++.h>
using namespace std;
int transitionPoint(int arr[], int n) {
    if(arr[n-1] == 0)
        return -1;
    
    if(n==1)
    {
        if(arr[0] == 1)
            return 0; 
        else
            return -1; 
    }
    int L = 0;
    int R = n-1;
    while(L <= R)
    {
        int mid = floor((L+R) /2);
        if(arr[mid] == 1 && arr[mid-1] != 1)
            return mid;
        else if(arr[mid] == 0)
            L = mid+1;
        else if(arr[mid] == 1)
            R = mid-1;
    }
    return -1;
}
int main()
{
  //write your code here
  int t;
  cin >> t;
  while(t-->0)
  {
    int a;
    cin >> a;
    int arr[a];
    for(int i=0;i<a;i++)
    {
     cin >> arr[i];
    }
    int ans = transitionPoint(arr, a);
    cout << ans << endl;
  }
  return 0;
}
